### DISCUSSION TITLE
How to integrate google ldap with keycloak

### DISCUSSION THREAD

**Discussion by iammuho** [keycloak/keycloak#12683] [Q&A]
Hello team,

I'm trying to integrate google LDAP with keycloak. I've created my LDAP client in GOOGLE ADMIN and downloaded the generated certificates and client credentials.

I've created a "federation provider" as describe in the documents but whatever I do, I'm getting below error;

```
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
	at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
	... 90 more
```

and also when I click the "test authentication" it always say "invalid".

here is how I've configure my ldap in keycloak

<img width="1283" alt="Screenshot 2022-06-23 at 14 14 52" src="https://user-images.githubusercontent.com/1788968/175296309-f83ba5ae-e426-4f30-b01e-1f63f758a56a.png">

<img width="1314" alt="Screenshot 2022-06-23 at 14 14 58" src="https://user-images.githubusercontent.com/1788968/175296315-3dbdd79a-c171-4abd-b279-1ab2b49ff4c4.png">

Can someone explain to me what should I do with the certificates from google and why my credentials say wrong. Btw, I've tested the ldap with credentials on my local, and it's working.

Btw, I'm using new keycloak operator in Kubernetes

**Comment by andreaTP** [comment]
Hi @iammuho , this discussion doesn't belong to the "New Kubernetes Operator", would you mind closing it and opening a relevant Issue if still necessary?

**Reply by iammuho** [reply]
Hey @andreaTP , I couldn't find how to close the discussion but moved to Q/A category. Do you have any idea how can I solve this issue ?

**Comment by romitkmehta** [comment]
Hey @iammuho and Keycloak community,
I am facing similar issue. I am getting the error as
```javax.naming.NoPermissionException: [LDAP: error code 50 - Insufficient Access Rights]; remaining name 'dc=<domain>,dc=com'```
I have been trying to sync users from Google workspace to Keycloak. I've enabled truststore and also have added required certificates in keystore.
Any insights on this?
